src/openstack_cloud/openstack_runner_manager.py

```diff
@@ -0,0 +1,708 @@
+                instance_id=instance.instance_id,
+                state=CloudRunnerState.from_openstack_server_status(instance.status),
+            )
+            for instance in instance_list
+        ]
+        if states is None:
+            return instance_list
+        return tuple(instance for instance in instance_list if instance.state in states)
+
+    def delete_runner(
+        self, instance_id: InstanceId, remove_token: str
+    ) -> runner_metrics.RunnerMetrics | None:
+        """Delete self-hosted runners.
+
+        Args:
+            instance_id: The instance id of the runner to delete.
+            remove_token: The GitHub remove token.
+
+        Returns:
+            Any metrics collected during the deletion of the runner.
+        """
+        instance = self._openstack_cloud.get_instance(instance_id)
+        if instance is None:
+            logger.warning(
+                "Unable to delete instance %s as it is not found",
+                self._openstack_cloud.get_server_name(instance_id),
+            )
+            return None
+
+        metric = runner_metrics.extract(
+            metrics_storage_manager=metrics_storage, runners=instance.server_name
+        )
+        self._delete_runner(instance, remove_token)
+        return next(metric, None)
+
+    def cleanup(self, remove_token: str) -> Iterator[runner_metrics.RunnerMetrics]:
+        """Cleanup runner and resource on the cloud.
+
+        Args:
+            remove_token: The GitHub remove token.
+
+        Returns:
+            Any metrics retrieved from cleanup runners.
+        """
+        runners = self._get_runner_health()
+        healthy_runner_names = [runner.server_name for runner in runners.healthy]
+        metrics = runner_metrics.extract(
+            metrics_storage_manager=metrics_storage, runners=set(healthy_runner_names)
+        )
+        for runner in runners.unhealthy:
+            self._delete_runner(runner, remove_token)
+
+        self._openstack_cloud.cleanup()
+        return metrics
+
+    def get_runner_health(self) -> RunnerByHealth:
+        """Get the runner health state.
+
+        Returns:
+            The runners by the health state.
+        """
+        runners = self._get_runner_health()
+        return RunnerByHealth(
+            tuple(runner.server_name for runner in runners.healthy),
+            tuple(runner.server_name for runner in runners.unhealthy),
+        )
+
+    def _delete_runner(self, instance: OpenstackInstance, remove_token: str) -> None:
+        """Delete self-hosted runners by openstack instance.
+
+        Args:
+            instance: The OpenStack instance.
+            remove_token: The GitHub remove token.
+        """
+        try:
+            ssh_conn = self._openstack_cloud.get_ssh_connection(instance)
+            self._pull_runner_metrics(instance.server_name, ssh_conn)
+
+            try:
+                OpenstackRunnerManager._run_github_runner_removal_script(
```

Comment:
Perhaps remove github from the function name

Reply:
Fixed.